### PR TITLE
aleph-one: Change livecheck

### PR DESCRIPTION
**Pull request by p-linnane**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [X] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [X] `brew audit --cask --online <cask>` is error-free.
- [X] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

**Comment by p-linnane** [comment]
See https://github.com/Homebrew/homebrew-cask/pull/132167#issuecomment-1255710488. This example utilizes `:github_latest`.

File: Casks/aleph-one.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@
   livecheck do
     url :url
     strategy :github_latest
```
Comment (samford) [inline Casks/aleph-one.rb:13]:
```suggestion
    url :homepage
```

File: Casks/aleph-one.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@
...
-    regex(%r{href=.*?/AlephOne-(\d+)-Mac\.dmg}i)
+    regex(%r{expanded_assets/release[._-](\d(?:\d+))}i)
```
Comment (samford) [inline Casks/aleph-one.rb:14]:
```suggestion
    regex(%r{href=.*?/AlephOne[._-]v?(\d+(?:\.\d+)*)[._-]Mac\.dmg}i)
```

**Review by samford — COMMENTED** [review]
We have a couple options in this particular case. The homepage links to the latest dmg file on GitHub, so we can easily check that. Otherwise, we could continue using the `GithubLatest` strategy by using a modified version of the default strategy regex that accounts for the `release-` prefix and date-based version format in tags (i.e., `%r{href=["']?[^"' >]*?/tag/release[._-]v?(\d+(?:\.\d+)*)["' >]}i`). That said, I've added suggestions that update the `livecheck` block to check the homepage (and bring the existing regex in line with prevailing standards).

In general, the `expanded_assets` URL simply uses the Git tag, so it's effectively no different from matching the version from the tag URL like the `GithubLatest` strategy does by default. I'll explain this a bit more in the related issue, though.

**Review by samford — APPROVED** [review]
Thanks!

**Pull request closed without merging**